test/wt_hang_analyzer/wt_hang_analyzer.py

```diff
@@ -0,0 +1,616 @@
+#!/usr/bin/env python
+"""Hang Analyzer module.
+
+A prototype hang analyzer for Evergreen integration to help investigate test timeouts
+
+1. Script supports taking dumps, and/or dumping a summary of useful information about a process
+2. Script will iterate through a list of interesting processes,
+    and run the tools from step 1. The list of processes can be provided as an option.
+
+Currenty only supports Linux. There are two issues with the MacOS and windows implementation:
+1. lldb cannot attach to processes in MacOS.
```

Comment:
If we have tickets made, can we mention the ticket numbers here?

I think it's a good practice to ensure that comments don't become stale. When working on a ticket, we can grep the ticket number and see if there are any comments that need to be amended.

Reply:
Updated the ticket number in the script.